Bug: Devcontainer linter failure - composites

Hello, I'm not actually sure this is isolated to the devcontainer, but i suspect it is based on previous conversations we've had.

```
github.com/alecthomas/chroma.Rule composite literal uses unkeyed fieldscomposites
```

Which appears to be caused by [this structure](https://github.com/lawrencegripper/azbrowse/blob/master/internal/pkg/views/itemview.go#L184) - specifically all 19 similar structures in the same file near that area.

![image](https://user-images.githubusercontent.com/1167891/70316523-d417db80-1813-11ea-8aaa-1951b3b3e87c.png)

I suspect, the `//nolint:govet` comments are supposed to be preventing this type of warning, but aren't working for some reason (I'm not familiar enough with this stuff to diagnose the issue). That said - One [potential solution](https://stackoverflow.com/a/54548495) seems to be to simply disable that type of warning globally, but I'd assume that isn't desirable for the project.

This is caused by [`vet`](https://golang.org/cmd/vet/)

```
✘-1 /workspaces/azbrowse [master ↓·2|✚ 1] 
12:08 # go vet ./internal/pkg/views/
# github.com/lawrencegripper/azbrowse/internal/pkg/views
internal/pkg/views/itemview.go:192:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:193:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:194:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:195:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:196:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:197:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:199:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:200:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:203:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:204:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:205:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:206:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:207:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:208:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:209:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:210:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:211:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:213:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
internal/pkg/views/itemview.go:216:5: github.com/alecthomas/chroma.Rule composite literal uses unkeyed fields
✘-2 /workspaces/azbrowse [master ↓·2|✚ 1] 
12:08 # go vet --composites=false ./internal/pkg/views/
```

As you can see, `--composites=false` fixes it. However, adding the following to the vscode settings still doesn't get the extension to ignore these:

```
"go.vetFlags": [
    "--composites=false"
]
```

which has me...very confused. Further, this error is __not__ present when the linter runs, at least via the shell.

--- a/azure-pipelines.yaml
+++ b/azure-pipelines.yaml
@@ -2,7 +2,15 @@ pool:
   vmImage: 'Ubuntu 18.04'
 
 steps:
-- script: sudo -E make devcontainer-release
+- script: |
+    # WAT? Yeah if the build doesn't have access to the variable it sets it to a litaral of the variable name with dollar sign and all
+    # this then causes make to fail so we find these and unset them. So GITHUB_TOKEN='$GITHUB_TOKEN' if we don't remove it. 
+    if [[ "$REASON" = "PullRequest" ]]; then
+      echo "Pull request so removing secret values which aren't set anyway but hey ho"
+      unset GITHUB_TOKEN
+      unset DOCKER_PASSWORD
+    fi
+    sudo -E make devcontainer-release
   displayName: 'Build, Test and *Publish *(only on master)'
   env:
     BUILD_NUMBER: $(Build.BuildId)
@@ -11,4 +19,5 @@ steps:
     GITHUB_TOKEN: $(GITHUB_TOKEN)
     IS_CI: 1
     PR_NUMBER: $(System.PullRequest.PullRequestNumber)
+    REASON: $(Build.Reason)
     BRANCH: $(Build.SourceBranchName)

--- a/internal/pkg/views/itemview.go
+++ b/internal/pkg/views/itemview.go
@@ -187,31 +187,31 @@ func configureYAMLHighlighting() {
 		chroma.Rules{
 			"root": {
 				chroma.Include("whitespace"),
-				{`#.*`, chroma.Comment, nil},                         //nolint:govet
-				{`!![^\s]+`, chroma.CommentPreproc, nil},             //nolint:govet
-				{`&[^\s]+`, chroma.CommentPreproc, nil},              //nolint:govet
-				{`\*[^\s]+`, chroma.CommentPreproc, nil},             //nolint:govet
-				{`^%include\s+[^\n\r]+`, chroma.CommentPreproc, nil}, //nolint:govet
-				{`([>|+-]\s+)(\s+)((?:(?:.*?$)(?:[\n\r]*?)?)*)`, chroma.ByGroups(chroma.StringDoc, chroma.StringDoc, chroma.StringDoc), nil}, //nolint:govet
-				chroma.Include("value"),                //nolint:govet
-				{`[?:,\[\]]`, chroma.Punctuation, nil}, //nolint:govet
-				{`.`, chroma.Text, nil},                //nolint:govet
+				{Pattern: `#.*`, Type: chroma.Comment, Mutator: nil},
+				{Pattern: `!![^\s]+`, Type: chroma.CommentPreproc, Mutator: nil},
+				{Pattern: `&[^\s]+`, Type: chroma.CommentPreproc, Mutator: nil},
+				{Pattern: `\*[^\s]+`, Type: chroma.CommentPreproc, Mutator: nil},
+				{Pattern: `^%include\s+[^\n\r]+`, Type: chroma.CommentPreproc, Mutator: nil},
+				{Pattern: `([>|+-]\s+)(\s+)((?:(?:.*?$)(?:[\n\r]*?)?)*)`, Type: chroma.ByGroups(chroma.StringDoc, chroma.StringDoc, chroma.StringDoc), Mutator: nil},
+				chroma.Include("value"),
+				{Pattern: `[?:,\[\]]`, Type: chroma.Punctuation, Mutator: nil},
+				{Pattern: `.`, Type: chroma.Text, Mutator: nil},
 			},
 			"value": {
-				{chroma.Words(``, `\b`, "true", "false", "null"), chroma.KeywordConstant, nil},                       //nolint:govet
-				{`"(?:\\.|[^"])*"`, chroma.StringDouble, nil},                                                        //nolint:govet
-				{`'(?:\\.|[^'])*'`, chroma.StringSingle, nil},                                                        //nolint:govet
-				{`\d\d\d\d-\d\d-\d\d([T ]\d\d:\d\d:\d\d(\.\d+)?(Z|\s+[-+]\d+)?)?`, chroma.LiteralDate, nil},          //nolint:govet
-				{`\b[+\-]?(0x[\da-f]+|0o[0-7]+|(\d+\.?\d*|\.?\d+)(e[\+\-]?\d+)?|\.inf|\.nan)\b`, chroma.Number, nil}, //nolint:govet
-				{`\b([\w]+)([ \t]*)([:]+)([ \t]*)(\d+\.?\d*|\.?\d+)(\s)`, chroma.ByGroups(chroma.Text, chroma.Whitespace, chroma.Punctuation, chroma.Whitespace, chroma.Number, chroma.Whitespace), nil}, //nolint:govet
-				{`\b([\w]+)([ \t]*)([:]+)([ \t]*)(true)\b`, chroma.ByGroups(chroma.Text, chroma.Whitespace, chroma.Punctuation, chroma.Whitespace, chroma.LiteralStringBoolean), nil},                    //nolint:govet
-				{`\b([\w]+)([ \t]*)([:]+)([ \t]*)(false)\b`, chroma.ByGroups(chroma.Text, chroma.Whitespace, chroma.Punctuation, chroma.Whitespace, chroma.LiteralStringBoolean), nil},                   //nolint:govet
-				{`\b([\w]+)([ \t]*)([:]+)([ \t]*)([\w\./\-_]+)\b`, chroma.ByGroups(chroma.Text, chroma.Whitespace, chroma.Punctuation, chroma.Whitespace, chroma.StringDouble), nil},                     //nolint:govet
-				// {`\b(?:[\w]+)(?:[:\b]+)(?:[\w+])\b`, ByGroups(KeywordReserved, Punctuation, StringDouble), nil},
-				{`\b[\w]+\b`, chroma.Text, nil}, //nolint:govet
+				{Pattern: chroma.Words(``, `\b`, "true", "false", "null"), Type: chroma.KeywordConstant, Mutator: nil},
+				{Pattern: `"(?:\\.|[^"])*"`, Type: chroma.StringDouble, Mutator: nil},
+				{Pattern: `'(?:\\.|[^'])*'`, Type: chroma.StringSingle, Mutator: nil},
+				{Pattern: `\d\d\d\d-\d\d-\d\d([T ]\d\d:\d\d:\d\d(\.\d+)?(Z|\s+[-+]\d+)?)?`, Type: chroma.LiteralDate, Mutator: nil},
+				{Pattern: `\b[+\-]?(0x[\da-f]+|0o[0-7]+|(\d+\.?\d*|\.?\d+)(e[\+\-]?\d+)?|\.inf|\.nan)\b`, Type: chroma.Number, Mutator: nil},
+				{Pattern: `\b([\w]+)([ \t]*)([:]+)([ \t]*)(\d+\.?\d*|\.?\d+)(\s)`, Type: chroma.ByGroups(chroma.Text, chroma.Whitespace, chroma.Punctuation, chroma.Whitespace, chroma.Number, chroma.Whitespace), Mutator: nil},
+				{Pattern: `\b([\w]+)([ \t]*)([:]+)([ \t]*)(true)\b`, Type: chroma.ByGroups(chroma.Text, chroma.Whitespace, chroma.Punctuation, chroma.Whitespace, chroma.LiteralStringBoolean), Mutator: nil},
+				{Pattern: `\b([\w]+)([ \t]*)([:]+)([ \t]*)(false)\b`, Type: chroma.ByGroups(chroma.Text, chroma.Whitespace, chroma.Punctuation, chroma.Whitespace, chroma.LiteralStringBoolean), Mutator: nil},
+				{Pattern: `\b([\w]+)([ \t]*)([:]+)([ \t]*)([\w\./\-_]+)\b`, Type: chroma.ByGroups(chroma.Text, chroma.Whitespace, chroma.Punctuation, chroma.Whitespace, chroma.StringDouble), Mutator: nil},
+				// {Pattern: `\b(?:[\w]+)(?:[:\b]+)(?:[\w+])\b`, Type: ByGroups(KeywordReserved, Punctuation, StringDouble), Mutator: nil},
+				{Pattern: `\b[\w]+\b`, Type: chroma.Text, Mutator: nil},
 			},
 			"whitespace": {
-				{`\s+`, chroma.Whitespace, nil}, //nolint:govet
+				{Pattern: `\s+`, Type: chroma.Whitespace, Mutator: nil},
 			},
 		},
 	)

